inbox/mailsync/backends/imap/generic.py

```diff
@@ -830,9 +832,17 @@ def condstore_refresh_flags(self, crispin_client):
                 interim_highestmodseq = max(v.modseq for k, v in flag_batch)
                 self.highestmodseq = interim_highestmodseq
 
+        del changed_flags  # free memory as soon as possible
+
+        remote_uids = crispin_client.all_uids()
+
         with session_scope(self.namespace_id) as db_session:
             local_uids = common.local_uids(self.account_id, db_session, self.folder_id)
-            expunged_uids = set(local_uids).difference(remote_uids)
+
+        expunged_uids = local_uids.difference(remote_uids)
```

Comment:
There was a `set(local_uids)` call here when this was already a `set`. This created extra copy in memory duplicating memory pressure here.